app/models/vm.rb

```diff
@@ -1,6 +1,7 @@
 class Vm < VmOrTemplate
   default_scope { where(:template => false) }
-
+  has_one :container_deployment, :through => :container_deployment_node
+  has_one :container_deployment_node
```

Comment:
@blomquisg yes it is.
To be more precise a vm can serve as a node/master inside of an OpenShift deployment, those masters/nodes are represented by ContainerDeploymentNode.
Those ContainerDeploymentNodes can represent a real vm that is managed by us (hence the connection here) or unmanaged by us and then we save the ip_or_hostname and we wont have a vm attached to it.